### PR TITLE
add function to retrieve commits from project id list

### PR DESCRIPTION
**Pull request by Dagg-h**
Add function to retrieve commits from gitlab API.

### CODE REVIEW

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 1]
```diff
@@ -93,7 +95,7 @@ def paginate_through_resource(
 
         while current_page < total_pages:
             page_variables = {"per_page": per_page, "current_page": current_page}
-            url = self.format_url(resource_api, page_variables, simple, order_by, since)
+            url = self.format_url(resource_api, page_variables, simple, order_by, since, sort)
```
Comment (Shrikey) [inline wherescape/connectors/gitlab/gitlab_wrapper.py]: The order of these attributes is wrong.

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 2]
```diff
@@ -231,3 +233,43 @@ def get_merge_requests(self):
...
+        keys_to_keep = COLUMN_NAMES_AND_DATA_TYPES["branches"].keys()
+
+        for project_id in project_ids:
+            resource_api = f"projects/{project_id}/repository/branches"
+            project_branches = self.paginate_through_resource(
+                resource_api, keys_to_keep, since=self.since, order_by="default", sort=False
+            )
+            all_branches.extend(project_branches)
+            
+        return all_branches
```
Comment (Shrikey) [inline wherescape/connectors/gitlab/gitlab_wrapper.py]: New line at the end

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ def make_request(self, url, method, payload={}):
         response = requests.request(method, url, data=payload, headers=headers)
         return response
 
-    def format_url(self, resource_api, page_variables, simple, order_by, since=None):
+    def format_url(self, resource_api, page_variables, simple, order_by, sort, since=None):
```
Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py:42]: You're adding a parameter without making it optional. Are you sure that this function is called nowhere else?

Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py:42]: You should add the `sort` parameter to the docstring of the function

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 4]
```diff
@@ -67,6 +68,7 @@ def paginate_through_resource(
         per_page=50,
         simple="false",
         order_by="id",
+        sort=True
```
Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py:71]: parameter must be added to docstring

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 5]
```diff
@@ -231,3 +233,44 @@ def get_merge_requests(self):
             all_merge_requests.extend(project_merge_requests)
 
         return all_merge_requests
+
+    def get_commits(self, project_ids):
+        """Get commits
```
Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py:238]: _strongly_ suggest to install pydocstyle & pylint in vscode

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 6]
```diff
@@ -231,3 +233,44 @@ def get_merge_requests(self):
             all_merge_requests.extend(project_merge_requests)
 
         return all_merge_requests
+
+    def get_commits(self, project_ids):
```
Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py:237]: input parameter missing in docstring

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 7]
```diff
@@ -231,3 +233,44 @@ def get_merge_requests(self):
...
+        """Get commits
+
+        Returns:
+        List of tuples with the commits values from the API
```
Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py:241]: commit values

File: wherescape/connectors/gitlab/gitlab_wrapper.py (hunk 5 above)
Comment (BFJonk) [reply wherescape/connectors/gitlab/gitlab_wrapper.py:238]: (docstring format tot conforming to linting standard)

File: wherescape/connectors/gitlab/gitlab_wrapper.py [hunk 8]
```diff
@@ -231,3 +233,44 @@ def get_merge_requests(self):
...
+        Returns:
+        List of tuples with the branches of the specific projects from the API
+        """
+        all_branches = []
+        keys_to_keep = COLUMN_NAMES_AND_DATA_TYPES["branches"].keys()
+
+        for project_id in project_ids:
+            resource_api = f"projects/{project_id}/repository/branches"
+            project_branches = self.paginate_through_resource(
+                resource_api, keys_to_keep, since=self.since, order_by="default", sort=False
```
Comment (BFJonk) [inline wherescape/connectors/gitlab/gitlab_wrapper.py:271]: why not specify the overwrite here?

**Pull request merged**